common/comms/tests/test_proton_queue_adaptor.py

```diff
@@ -0,0 +1,155 @@
+"""Module for testing the Proton queue adaptor functionality."""
+import asyncio
+import unittest.mock
+
+import proton
+
+import comms.proton_queue_adaptor
+import utilities.test_utilities
+
+TEST_MESSAGE = "TEST MESSAGE"
+TEST_PROTON_MESSAGE = unittest.mock.Mock()
+TEST_QUEUE_HOST = "TEST QUEUE HOST"
+TEST_EXCEPTION = Exception()
+TEST_SIDE_EFFECT = unittest.mock.Mock(side_effect=TEST_EXCEPTION)
+
+
+class TestProtonQueueAdaptor(unittest.TestCase):
+    """Class to contain tests for the ProtonQueueAdaptor functionality."""
+
+    def setUp(self) -> None:
+        self.service = comms.proton_queue_adaptor.ProtonQueueAdaptor(host=TEST_QUEUE_HOST)
+
+    # TESTING SEND ASYNC METHOD
+    @unittest.mock.patch.object(proton.reactor, "Container")
+    @utilities.test_utilities.async_test
+    async def test_send_async_success(self, mock_container):
+        """Test happy path of send_async."""
+        mock_container.run.return_value = asyncio.Future()
+        await self.service.send_async(TEST_MESSAGE)
```

Comment:
Feels like for this sort of test you'd want to use the `AsyncTestCase` provided by Tornado, as you're using the Tornado `run_in_executor` method.

Also, these tests aren't asserting that anything has been done.

Reply:
unfortunately, as we rely on proton to provide the actual functionality, and the only modified state would be external to the application there is very little to assert other than the container has been run.